lib/util/validationError.ts

```diff
@@ -24,10 +24,38 @@ export class ValidationError {
   public constructor(public readonly name: string, public readonly severity: Severity) {}
 }
 
-const validationErrorEntry = (id: string, severity: Severity): [string, ValidationError] => [
-  id,
-  new ValidationError(id, severity)
-]
+const validationErrorEntry = (
+  id: ExtendedErrorCode | string,
+  severity: Severity
+): [string, ValidationError] => [id, new ValidationError(id, severity)]
+
+export type ExtendedErrorCode = ErrorCode & WrapperErrorCode
+
+export type ErrorCode =
```

Comment:
You can declare a constant and then use `typeof yourConstant` and `keyof` to declare a type.